### PR TITLE
bug(query): Fix for issues in metadata query capability

### PR DESCRIPTION
**Pull request by vishramachandran**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [ ] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

* SEGV was seen when invoking promQL time series API because of
  prettyPrint regression
* LabelValues API should not accept promQL matcher. That can be
  supported later.
* Earlier code was not skipping metadata for evicted partitions properly
* Default lookback time was not sufficient

### CODE REVIEW

File: cli/src/main/scala/filodb.cli/CliMain.scala [hunk 1]
```diff
@@ -392,13 +383,13 @@ object CliMain extends ArgMain[Arguments] with CsvImportExport with FilodbCluste
         try {
           client.logicalPlan2Query(ref, plan, qOpts) match {
             case QueryResult(_, schema, result) => {
-              println(s"Number of Range Vectors: ${result.size}")
-              result.foreach(rv => println(rv.prettyPrint(schema)))
-            }
-            case QueryError(_,ex)               => println(s"ERROR: ${ex.getMessage}")
+                                                      println(s"Number of Range Vectors: ${result.size}")
+                                                      result.foreach(rv => println(rv.prettyPrint(schema)))
+                                                   }
```
Comment (velvia) [inline cli/src/main/scala/filodb.cli/CliMain.scala]: Can we please remove the braces?

File: coordinator/src/main/scala/filodb.coordinator/queryengine2/QueryEngine.scala [hunk 2]
```diff
@@ -198,10 +198,18 @@ class QueryEngine(dataset: Dataset,
...
                                       options: QueryOptions,
                                       lp: LabelValues): Seq[LabelValuesExec] = {
-    shardsFromFilters(lp.filters, options).map { shard =>
+    val filters = lp.labelConstraints.map { case (k, v) =>
+      new ColumnFilter(k, Filter.Equals(v))
+    }.toSeq
+    val shardsToHit = if (shardColumns.toSet.subsetOf(lp.labelConstraints.keySet)) {
+                        shardsFromFilters(filters, options)
+                      } else {
+                        shardMapperFunc.assignedShards
```
Comment (velvia) [inline coordinator/src/main/scala/filodb.coordinator/queryengine2/QueryEngine.scala]: I wonder if this should be in FiloDB itself or an external service (just pass along all of the shards there as an override).

File: core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala [hunk 3]
```diff
@@ -474,13 +474,18 @@ class TimeSeriesShard(val dataset: Dataset,
...
-      dataset.partKeySchema.consumeMapItems(nextPart.partKeyBase, nextPart.partKeyOffset, 0, consumer)
-      result ++= consumer.labelValues
+      val partId = partIterator.next()
+      val nextPart = getPartitionFromPartId(partId)
+      if (nextPart != UnsafeUtils.ZeroPointer) {
+        val consumer = new SeqIndexValueConsumer(indexName)
+        // FIXME This is non-performant and temporary fix for fetching label values based on filter criteria.
+        // Other strategies needs to be evaluated for making this performant - create facets for predefined fields or
+        // have a centralized service/store for serving metadata
+        dataset.partKeySchema.consumeMapItems(nextPart.partKeyBase, nextPart.partKeyOffset, 0, consumer)
```
Comment (velvia) [inline core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala]: This needs to be almost immediately fixed or else non-Prometheus schemas will break, in a very ugly and non obvious way.  (ie possibly SEGV's or hangs).

Comment (vishramachandran) [reply core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala]: Agreed.

**Pull request merged**